parquet-tools/src/test/java/org/apache/parquet/tools/command/TestPruneColumnsCommand.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.tools.command;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.PosixParser;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.format.converter.ParquetMetadataConverter;
+import org.apache.parquet.hadoop.ParquetFileReader;
+import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.Type;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Random;
+
+import static org.junit.Assert.assertEquals;
+
+public class TestPruneColumnsCommand {
+
+  private Configuration conf = new Configuration();
+  private PruneColumnsCommand command = new PruneColumnsCommand();
+  private Random rand = new Random();
+
+  @Test
+  public void testPruneOneColumn() throws Exception {
+    // File test.parquet has 3 columns: "name", "age", "gender"
+    String inputFile = "src/test/java/org/apache/parquet/tools/data/test.parquet";
+    String outputFile =  rand.nextInt() + ".parquet";
+    String cargs[] = {inputFile, outputFile, "gender"};
+    executeCommandLine(cargs);
+    ParquetMetadata pmd = ParquetFileReader.readFooter(conf, new Path(outputFile), ParquetMetadataConverter.NO_FILTER);
+    MessageType schema = pmd.getFileMetaData().getSchema();
+    List<Type> fields = schema.getFields();
+    assertEquals(fields.size(), 2);
+    assertEquals(fields.get(0).getName(), "name");
+    assertEquals(fields.get(1).getName(), "age");
```

Comment:
It would also be great to verify the actual data. Maybe, you may use the `dump` command?

Reply:
Yes. I just wrote a reader to verify.